packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/SpanDetail/KeyValuesTable.js

```diff
@@ -30,27 +31,75 @@ function parseIfJson(value) {
   return value;
 }
 
+function markupAsDiv(markup) {
+  // eslint-disable-next-line react/no-danger
+  return <div dangerouslySetInnerHTML={{ __html: markup }} />;
+}
+
+function markupAsSpan(markup) {
+  const spanMarkup = markup.replace(/^<div /i, '<span ').replace(/<\/div>$/i, '</span>');
+  // eslint-disable-next-line react/no-danger
+  return <span dangerouslySetInnerHTML={{ __html: spanMarkup }} />;
+}
```

Comment:
Instead of manipulating the HTML seems like adding `ub-inline-block` to the `<div>` would work?

```js
            const jsonTable = (
              // eslint-disable-next-line react/no-danger
              <div className="ub-inline-block" dangerouslySetInnerHTML={{ __html: jsonMarkup(parseIfJson(row.value)) }} />
            );
```

I think that might also need

```css
.KeyValueTable--body > tr > td {
    padding: 0.25rem 0.5rem;
    vertical-align: top;
}
```

Reply:
Thank you for your suggestion. I have added a commit with this change.